### PR TITLE
Support anonymous blocks for new Oracle dialect

### PR DESCRIPTION
**Pull request by rathboma**
Oracle supports statement-level blocks (because...reasons)
This adds support for them.
It also adds oracle as a dialect

### CODE REVIEW

**Comment by rathboma** [comment]
@MasterOdin be good to get eyes on this briefly before merging, but it's working. Going to use in BKS either way

File: test/identifier/multiple-statement.spec.ts [hunk 1]
```diff
@@ -127,6 +127,63 @@ describe('identifier', () => {
       expect(actual).to.eql(expected);
     });
 
+    describe('identifying statements with anonymous blocks', () => {
+      it('should identify a create table then a block', () => {
```
Comment (MasterOdin) [inline test/identifier/multiple-statement.spec.ts:131]: why have this test repeated here and in `parser/oracle.spec.ts`?

File: test/parser/oracle.spec.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+          FROM
+            example_table;
+        END
+        `;
+      const result = parse(sql, false, 'oracle');
+
+      expect(result.body[0].type).to.eq('ANON_BLOCK');
+      expect(result.body[0].start).to.eq(0);
+      expect(result.body[0].end).to.eq(120);
+      expect(result.body.length).to.eql(2);
```
Comment (MasterOdin) [inline test/parser/oracle.spec.ts:48]: I think this should be the first assert where if there were no statements found, then we'd get an `undefined` error for a test failure, which isn't as meaningful imo.

**Review by MasterOdin — CHANGES_REQUESTED** [review]
For all the tests in `test/parser/oracle.spec.ts`, I think it would be good to have test:
* Number of statements
* Type of each statement
* Start of each statement
* End of each statement

So that it's nice and uniform, and we're testing the full scope of things.

File: test/identifier/multiple-statement.spec.ts (hunk 1 above)
Comment (rathboma) [reply test/identifier/multiple-statement.spec.ts:131]: Just as an idiot check to be honest. Costs very little, gives me some assurances

File: test/parser/oracle.spec.ts (hunk 2 above)
Comment (rathboma) [reply test/parser/oracle.spec.ts:48]: Yes I reorganized this incorrectly, moving the other way around

**Comment by rathboma** [comment]
I'll address these couple of issues, then I'll merge it.

**Comment by rathboma** [comment]
@MasterOdin I've added a few changes, including adding a new execution type to account for the fact that ANON_BLOCKs should be identifiable in strict mode, and that their execution type is unknown.

I think introducing a new execution type is cleaner than muddying the meaning of `UNKNOWN`.

I don't think this change requires a major version bump, as the new execution and statement types are only valid for Oracle, so no behavior has changed for any other databases.

**Comment by rathboma** [comment]
@MasterOdin I'll merge this tomorrow if you don't have any other changes.

**Review by MasterOdin — APPROVED** [review]

**Pull request merged**